src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,185 @@
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const double numClass,
+    const size_t maxIter,
+    const double tol) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    numClass(numClass)
+{
+  numClassifier = numClass * ((numClass - 1) / 2);
+  Train(data, labels, maxIter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const double regularization,
+    const bool fitIntercept,
+    const double numClass) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    numClass(numClass)
+{
+  numClassifier = numClass * (numClass - 1) / 2;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t maxIter,
+    const double tol)
+{
+  classesClassifier = arma::zeros(numClassifier, 2);
+  double val = 0;
+
+  size_t countClass = 0;
+  for (size_t i = 0; i < numClass; i++)
+  {
+    for (size_t j = i+1; j < numClass; j++)
+    {
+      classesClassifier(countClass, 0) = i;
+      classesClassifier(countClass, 1) = j;
+      arma::rowvec tempLables = arma::zeros(1, data.n_cols);
+      for (size_t k = 0; k < data.n_cols; k++)
+      {
+        if (labels(k) == j)
+          tempLables(k) = -1;
+        if (labels(k) ==  i)
+          tempLables(k) = 1;
+      }
+
+      KernelSVMFunction<MatType,
+                        KernelType> svm(
+                          data.cols(
+                            arma::find(tempLables == 1 || tempLables == -1)),
+                          tempLables.cols(
+                            arma::find(tempLables == 1 || tempLables == -1)),
+                          regularization, fitIntercept, maxIter, tol);
+      network.push_back(svm);
+
+      countClass++;
+      val++;
+    }
+  }
+  return val;
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels) const
+{
+  arma::mat scores;
+  Classify(data, labels, scores);
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels,
+    arma::mat& scores) const
+{
```

Comment:
While implementing this function I am using multiple `for`. If you have any better suggestion please let me know I can remove them.